src/music/extmidi.cpp

```diff
@@ -130,23 +130,32 @@ void MusicDriver_ExtMidi::DoPlay()
 	}
 }
 
-void MusicDriver_ExtMidi::DoStop()
+static bool DoKillWait(pid_t &pid, int signal)
 {
-	if (this->pid <= 0) return;
-
-	/* First try to gracefully stop for about five seconds;
-	 * 5 seconds = 5000 milliseconds, 10 ms per cycle => 500 cycles. */
 	for (int i = 0; i < 500; i++) {
-		kill(this->pid, SIGTERM);
-		if (waitpid(this->pid, nullptr, WNOHANG) == this->pid) {
+		kill(pid, signal);
+		if (waitpid(pid, nullptr, WNOHANG) == pid) {
 			/* It has shut down, so we are done */
-			this->pid = -1;
-			return;
+			pid = -1;
+			return true;
 		}
 		/* Wait 10 milliseconds. */
 		CSleep(10);
 	}
 
+	return false;
+}
+
+void MusicDriver_ExtMidi::DoStop()
+{
+	if (this->pid <= 0) return;
+
+	/* First try to gracefully stop for about five seconds;
+	 * 5 seconds = 5000 milliseconds, 10 ms per cycle => 500 cycles. */
```

Comment:
Not sure how to solve this exactly, but... with this change the magic 500 and 10 in `DoKillWait` are not documented in the function, but rather here even though they are essentially implementation details of how to get to the 5 seconds. On the other hand, writing down it is going to gracefully try to interrupt, then terminate and if all else fails kill the driver belongs to this function. So I guess the comment ought to be split over the two functions.